rasa/core/actions/action.py

```diff
@@ -406,7 +405,7 @@ def get_full_retrieval_name(
             return None
 
         response_selector_properties = latest_message.parse_data[
-            RESPONSE_SELECTOR_PROPERTY_NAME
+            RESPONSE_SELECTOR_PROPERTY_NAME  # type: ignore[misc]
```

Comment:
`parse_data` is defined as a `TypedDict`, but `mypy` doesn't support variables as keys of a `TypedDict` (both in definition and access -- I've tried using `Final` and `Literal` without any luck). I made an attempt to use a dataclass but stopped halfway as I think it requires a proper design phase and understanding of the problem first. I can open a separate issue to document my findings (or non-findings).